doc/howto/container_gpu_passthrough_with_docker.md

```diff
@@ -0,0 +1,145 @@
+(container-gpu-passthrough-with-docker)=
+# How to pass an NVIDIA GPU to a container
+
+If you have an NVIDIA GPU (either discrete (dGPU) or integrated (iGPU)) and you want to pass the runtime libraries and configuration installed on your host to your container, you should add a {ref}`LXD GPU device <devices-gpu>`.
+Consider the following scenario:
+
+Your host is an NVIDIA single board computer that has a Tegra SoC with an iGPU, and you have the Tegra SDK installed on the host. You want to create a LXD container and run an application inside the container using the iGPU as a compute backend. You want to run this application inside a Docker container (or another OCI-compliant runtime).
+To achieve this, complete the following steps:
+
+1. Initialize a LXD container with {config:option}`instance-security:security.nesting` enabled (needed for running a Docker container inside a LXD container):
+
+        lxc init ubuntu:24.04 t1 --config security.nesting=true
```

Comment:
If the container is initialised this way, the docker container later on will use the `vfs` storage driver which is very space inefficient and requires more than 100GB of space for downloading the container image.

After discussion with @tomponline we found that creating a `btrfs` storage pool, drastically reduces the space needed to run the container. This would also be automatically picked up by docker.

Therefore in my opinion, step 1 should be the creation of the storage pool:
```
lxc storage create p1 btrfs size=15GiB
```

And then use it for initialising the container:
```
lxc init ubuntu:24.04 t1 --config security.nesting=true -s p1
```

Reply:
agreed, or suggest using the syscall interception mode, as per our docs (this should be tested to check if works as it would mean one could use a dir pool without needing to setup a new btrfs pool).

Reply:
@DocSepp @tomponline updated

Reply:
Sorry @tomponline (I updated my earlier message), I actually face the same issue as @DocSepp for the ARM board.

Here is the `docker info` inside the LXD container for my x86 worstation:
```bash 
Client: Docker Engine - Community
 Version:    27.1.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 27.1.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 nvidia runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fc6bcff51318944179630522a095cc9dbf9f353
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.8.0-41-generic
 Operating System: Ubuntu 24.04 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 24
 Total Memory: 31.18GiB
 Name: t1
 ID: 0b4d9cd6-397d-4563-a2b6-45389cb714a9
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

and for the IGX board:
```bash
Client: Docker Engine - Community
 Version:    27.1.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 27.1.2
 Storage Driver: vfs
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 nvidia
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fc6bcff51318944179630522a095cc9dbf9f353
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.15.0-1012-nvidia-tegra-igx
 Operating System: Ubuntu 24.04 LTS
 OSType: linux
 Architecture: aarch64
 CPUs: 12
 Total Memory: 53.51GiB
 Name: t1
 ID: e5abe472-90f4-4117-b8bf-fcce731c5c93
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

____


I try to enable `overlay2` inside the LXD container on the IGX board (following [this](https://docs.docker.com/engine/storage/drivers/overlayfs-driver/#prerequisites)) but it fails

Reply:
Do you see something unusual in the `docker info` from the board?

Reply:
Maybe that system's kernel is missing overlay module.

Ok let's document both options then

Reply:
fixed